drivers/watchdog/wdt_nxp_fs26.c

```diff
@@ -0,0 +1,838 @@
+		break;
+	case WDT_FLAG_RESET_NONE:
+		data->fs_reaction = WD_FS_REACTION_NO_ACTION >> WD_FS_REACTION_SHIFT;
+		LOG_DBG("Configuring non-reset mode");
+		break;
+	default:
+		LOG_ERR("Unsupported watchdog configuration flag");
+		return -EINVAL;
+	}
+
+	data->callback = cfg->callback;
+	data->timeout_installed = true;
+
+	/* Always return channel ID equal to 0 */
+	return 0;
+}
+
+static int wdt_nxp_fs26_disable(const struct device *dev)
+{
+	const struct wdt_nxp_fs26_config *config = dev->config;
+	struct wdt_nxp_fs26_data *data = dev->data;
+	struct fs26_spi_rx_frame rx_frame;
+	uint32_t regval;
+
+	if (fs26_getreg(&config->spi, FS26_FS_WDW_DURATION, &rx_frame)) {
+		return -EIO;
+	}
+	if ((rx_frame.data & WDW_PERIOD_MASK) == WDW_PERIOD_DISABLE) {
+		LOG_ERR("Watchdog already disabled");
+		return -EFAULT;
+	}
+
+	/* The watchdog window can be disabled only during the initialization phase */
+	if (fs26_goto_init_fs_state(dev)) {
+		LOG_ERR("Failed to go to INIT_FS");
+		return -EIO;
+	}
+
+	regval = WDW_PERIOD_DISABLE | WDW_RECOVERY_DISABLE;
+	fs26_setreg(&config->spi, FS26_FS_WDW_DURATION, regval);
+	fs26_setreg(&config->spi, FS26_FS_NOT_WDW_DURATION, ~regval);
+
+	/* The watchdog disabling is effective when the initialization phase is closed */
+	if (fs26_exit_init_fs_state(dev)) {
+		LOG_ERR("Failed to close INIT_FS");
+		return -EIO;
+	}
+
+	LOG_DBG("Watchdog disabled");
+	data->timeout_installed = false;
+
+	return 0;
+}
+
+static void wdt_nxp_fs26_int_thread(const struct device *dev)
+{
+	const struct wdt_nxp_fs26_config *config = dev->config;
+	struct wdt_nxp_fs26_data *data = dev->data;
+	struct fs26_spi_rx_frame rx_frame;
+	uint32_t regval;
+
+	while (1) {
+		k_sem_take(&data->int_sem, K_FOREVER);
+
+		if ((!fs26_getreg(&config->spi, FS26_FS_GRL_FLAGS, &rx_frame))
+			&& ((rx_frame.data & FS_WD_G_MASK) == FS_WD_G)) {
+
+			if ((!fs26_getreg(&config->spi, FS26_FS_DIAG_SAFETY1, &rx_frame))
+				&& (rx_frame.data & BAD_WD_TIMING)) {
+
+				/* Clear flag */
+				regval = BAD_WD_TIMING;
+				fs26_setreg(&config->spi, FS26_FS_DIAG_SAFETY1, regval);
+
+				/* Invoke user callback */
+				if (data->callback && data->timeout_installed) {
+					data->callback(dev, 0);
+				}
+			}
+		}
+	}
+}
+
+static void wdt_nxp_fs26_int_callback(const struct device *dev,
+				      struct gpio_callback *cb,
+				      uint32_t pins)
+{
+	struct wdt_nxp_fs26_data *data = CONTAINER_OF(cb, struct wdt_nxp_fs26_data,
+						      int_gpio_cb);
+
+	ARG_UNUSED(dev);
+	ARG_UNUSED(pins);
+
+	k_sem_give(&data->int_sem);
+}
+
+static int wdt_nxp_fs26_init(const struct device *dev)
+{
+	const struct wdt_nxp_fs26_config *config = dev->config;
+	struct wdt_nxp_fs26_data *data = dev->data;
+	struct fs26_spi_rx_frame rx_frame;
+	uint32_t regval;
+
+	/* Validate bus is ready */
+	if (!spi_is_ready_dt(&config->spi)) {
+		return -ENODEV;
+	}
+
+	k_sem_init(&data->int_sem, 0, 1);
+
+	/* Configure GPIO used for INTB signal */
+	if (!device_is_ready(config->int_gpio.port)) {
```

Comment:
Just a suggestion here: you can use `gpio_is_ready_dt` instead of `device_is_ready` for GPIO DT spec structs

Reply:
updated as suggested